### ISSUE TITLE
Abort in service::storage_service::handle_state_normal in test_decommission_after_changing_node_ip

### ISSUE THREAD

**Issue by asias** [scylladb/scylladb#21930]
Observed in CI:

```
https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13404/artifact/logs-full.release.001/1734085322685_update_cluster_layout_tests.py%3A%3ATestUpdateClusterLayout%3A%3Atest_decommission_after_changing_node_ip/
```

Build and packages:
```
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/13805/artifact/00-Build.txt
timestamp: 2024-12-13T11:14:31Z
git@github.com:scylladb/scylla.git-sha: 21861
jenkins-job-path: scylla-master/scylla-ci
jenkins-job-number: 13805
url-id: 2024-12-13T11:14:31Z
reloc-pack-url: downloads.scylladb.com/unstable/scylla/master/relocatable/2024-12-13T11:14:31Z/
rpm-deb-pack-url: downloads.scylladb.com/unstable/scylla/master/build-packages/2024-12-13T11:14:31Z/
scylla-product: scylla
scylla-release: 0.20241213.007ef34a896c
scylla-version: 6.3.0~dev
scylla-x86_64-BuildID[sha1]: 0247924dc5ca2597ec1ac278bc02bb69ceecfd41
scylla-debug-x86_64-BuildID[sha1]: e417ff9d13686ce6c9ee5efb599b3bca0e2b5caf
unified-pack-url-x86_64: downloads.scylladb.com/unstable/scylla/master/relocatable/2024-12-13T11:14:31Z/scylla-unified-6.3.0~dev-0.20241213.007ef34a896c.x86_64.tar.gz
docker-image-name-x86_64: scylladb-ci:6.3.0-dev-0.20241213.007ef34a896c-x86_64
docker-image-name: scylladb-ci:6.3.0-dev-0.20241213.007ef34a896c
```

Bactraces:
```
bash-5.2$ seastar/scripts/seastar-addr2line -e 47924dc5ca2597ec1ac278bc02bb69ceecfd41.debug 
0x5e31064
0x5debfab
0x5e23bd8
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x40cff
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x99663
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x40c4d
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x28901
0x5dbe44c
0x3f5746c
0x401c57f
0x401bff0
0x3d38d1d
0x3f6175d
0x3f5ea79
0x3cd0cfe
0x3c5336a
0x3cfa4df
0x146142a
0x5dfe96f
0x5dfff3a
0x5e01117
0x5e004c8
0x5d8e043
0x5d8d3a3
0x1405427
0x14072b0
0x1402f63
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a087
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a14a
0x1400f14

[Backtrace #0]
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:69
seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:805
seastar::print_with_backtrace(char const*, bool) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:850
 (inlined by) seastar::sigabrt_action() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3995
  (inlined by) seastar::install_oneshot_signal_handler<6, (void (*)())(&seastar::sigabrt_action)>()::{lambda(int, siginfo_t*, void*)#1}::operator()(int, siginfo_t*, void*) const at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3972
   (inlined by) seastar::install_oneshot_signal_handler<6, (void (*)())(&seastar::sigabrt_action)>()::{lambda(int, siginfo_t*, void*)#1}::__invoke(int, siginfo_t*, void*) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3967
   addr2line: '/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6': No such file
   /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x40cff 
   /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x99663 
   /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x40c4d 
   /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x28901 
   seastar::on_fatal_internal_error(seastar::logger&, std::basic_string_view<char, std::char_traits<char> >) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/on_internal_error.cc:80
   service::storage_service::handle_state_normal(gms::inet_address, utils::tagged_uuid<gms::permit_id_tag>) at ././service/storage_service.cc:2512
   service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0::operator()(gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>) const at ././service/storage_service.cc:2663
   seastar::future<void> std::__invoke_impl<seastar::future<void>, service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0&, gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag> >(std::__invoke_other, service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0&, gms::inet_address&&, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>&&) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:61
    (inlined by) std::enable_if<is_invocable_r_v<seastar::future<void>, service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0&, gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag> >, seastar::future<void> >::type std::__invoke_r<seastar::future<void>, service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0&, gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag> >(service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0&, gms::inet_address&&, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>&&) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:114
	 (inlined by) std::_Function_handler<seastar::future<void> (gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>), service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>)::$_0>::_M_invoke(std::_Any_data const&, gms::inet_address&&, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>&&) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:290
	 gms::i_endpoint_state_change_subscriber::on_application_state_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, gms::application_state, utils::tagged_uuid<gms::permit_id_tag>, std::function<seastar::future<void> (gms::inet_address, gms::versioned_value const&, utils::tagged_uuid<gms::permit_id_tag>)>) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:591
	 service::storage_service::on_change(gms::inet_address, std::unordered_map<gms::application_state, gms::versioned_value, std::hash<gms::application_state>, std::equal_to<gms::application_state>, std::allocator<std::pair<gms::application_state const, gms::versioned_value> > > const&, utils::tagged_uuid<gms::permit_id_tag>) at ././service/storage_service.cc:2651
	 non-virtual thunk to service::storage_service::on_join(gms::inet_address, seastar::lw_shared_ptr<gms::endpoint_state const>, utils::tagged_uuid<gms::permit_id_tag>) at ././service/storage_service.cc:2620
	 seastar::noncopyable_function<seastar::future<void> (seastar::shared_ptr<gms::i_endpoint_state_change_subscriber>)>::indirect_vtable_for<gms::gossiper::handle_major_state_change(gms::inet_address, gms::endpoint_state, utils::tagged_uuid<gms::permit_id_tag>, bool)::$_1>::call(seastar::noncopyable_function<seastar::future<void> (seastar::shared_ptr<gms::i_endpoint_state_change_subscriber>)> const*, seastar::shared_ptr<gms::i_endpoint_state_change_subscriber>) at ././gms/gossiper.cc:1837
	 atomic_vector<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> >::for_each(seastar::noncopyable_function<seastar::future<void> (seastar::shared_ptr<gms::i_endpoint_state_change_subscriber>)>) const at ././seastar/include/seastar/util/noncopyable_function.hh:215
	 gms::gossiper::handle_major_state_change(gms::inet_address, gms::endpoint_state, utils::tagged_uuid<gms::permit_id_tag>, bool) at ././gms/gossiper.cc:1836
	 std::__n4861::coroutine_handle<seastar::internal::coroutine_traits_base<void>::promise_type>::resume() const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/coroutine:242
	  (inlined by) seastar::internal::coroutine_traits_base<void>::promise_type::run_and_dispose() at ././seastar/include/seastar/core/coroutine.hh:122
	  seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:2606
	  seastar::reactor::run_some_tasks() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3078
	  seastar::reactor::do_run() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3246
	  seastar::reactor::run() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3136
	  seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/app-template.cc:276
	  seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/app-template.cc:167
	  scylla_main(int, char**) at ././main.cc:778
	  std::function<int (int, char**)>::operator()(int, char**) const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:591
	  main at ././main.cc:2391
	  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x2a087 
	  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x2a14a 
	  _start at ??:?
```

Logs:

[1734085322685_update_cluster_layout_tests.py__TestUpdateClusterLayout__test_decommission_after_changing_node_ip.zip](https://github.com/user-attachments/files/18146096/1734085322685_update_cluster_layout_tests.py__TestUpdateClusterLayout__test_decommission_after_changing_node_ip.zip)

**Comment by bhalevy** [comment]
```
ERROR 2024-12-13 10:20:01,192 [shard 0: gms] storage_service - endpoint=127.0.64.3 is marked for removal but still owns 256 tokens, at: 0x630357e 0x63037b0 0x6303ab8 0x5dbe28e 0x5dbe447 0x3f5746c 0x401c57f 0x401bff0 0x3d38d1d 0x3f6175d 0x3f5ea79 0x3cd0cfe 0x3c5336a 0x3cfa4df 0x146142a 0x5dfe96f 0x5dfff3a 0x5e01117 0x5e004c8 0x5d8e043 0x5d8d3a3 0x1405427 0x14072b0 0x1402f63 /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a087 /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a14a 0x1400f14
```

@kbr-scylla mind looking into this?

**Comment by kostja** [comment]
@bhalevy is this gossip only mode? Is gossip or raft topology change in use?

**Comment by kbr-scylla** [comment]
Changing node IP test in gossip mode --- I suspect it is related to @gleb-cloudius' recent patches -- Gleb please investigate 


note it happened https://github.com/scylladb/scylladb/pull/21861 so it might be timing related somehow (enabling small table optimization speeds things up during topo changes)

The commit was 007ef34a896c6ecbb60c359fce70e379bf44d3a9 (merge commit in PR) based on 0114e4c2ae11bdcc6c7348692bef61d357403b6c from `master` branch

**Comment by asias** [comment]
It seems we saw it again even with the fix:

CI:
https://github.com/scylladb/scylladb/pull/21861#issuecomment-2560699907

Test:
https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14051/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest___test_decommission_after_changing_node_ip

```
        if critical_errors:
>           raise AssertionError(f"Critical errors found: {critical_errors}\nOther errors: {found_errors}")
E           AssertionError: Critical errors found: [('node3', ['Aborting on shard 0, in scheduling group gossip.'])]
E           Other errors: [('node3', ['ERROR 2024-12-24 03:19:07,812 [shard 0: gms] storage_service - endpoint=127.0.88.3 is marked for removal but still owns 256 tokens, at: 0x1443e8e 0x14440c0 0x14443c8 0x148ba7e 0x148bc37 0x4935f25 0x13c8afa 0x138f7cf 0x1390d9a 0x1391f77 0x1391328 0x1648153 0x16474b3 0x14ee377 0x14f0200 0x14ebeb3 /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a087 /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a14a 0x1365fc4'])]
```

The CI is using git cd2a2bd02143598cc6e88ad2b99f6ed7301355ba which contains 91cddcc17f7049bebaa0429e9a2d10ae56a9f4f3.

**Comment by asias** [comment]
Log:
https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13621/artifact/logs-full.release.001/1735010469128_update_cluster_layout_tests.py%3A%3ATestUpdateClusterLayout%3A%3Atest_decommission_after_changing_node_ip/node3.log

PKG:
https://downloads.scylladb.com/unstable/scylla/master/build-packages/2024-12-24T04:31:33Z/deb/

Backtraces:
```
bash-5.2$ seastar/scripts/seastar-addr2line -e ./tmp1/usr/lib/debug/.build-id/b7/eb44228088dfdb838df0e2f4d933982e8a4614.debug
0x13c5984
  0x137ce0b
  0x13b4668
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x40d0f
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x99bd3
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x40c5d
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x28901
  0x148bc3c
  0x4935f25
  0x13c8afa
  0x138f7cf
  0x1390d9a
  0x1391f77
  0x1391328
  0x1648153
  0x16474b3
  0x14ee377
  0x14f0200
  0x14ebeb3
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a087
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6+0x2a14a
  0x1365fc4

[Backtrace #0]
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/RelWithDebInfo/seastar/./seastar/include/seastar/util/backtrace.hh:69
seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:805
seastar::print_with_backtrace(char const*, bool) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:850
 (inlined by) seastar::sigabrt_action() at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3995
 (inlined by) seastar::install_oneshot_signal_handler<6, (void (*)())(&seastar::sigabrt_action)>()::{lambda(int, siginfo_t*, void*)#1}::operator()(int, siginfo_t*, void*) const at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3972
 (inlined by) seastar::install_oneshot_signal_handler<6, (void (*)())(&seastar::sigabrt_action)>()::{lambda(int, siginfo_t*, void*)#1}::__invoke(int, siginfo_t*, void*) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3967
addr2line: '/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6': No such file
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x40d0f 
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x99bd3 
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x40c5d 
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x28901 
seastar::on_fatal_internal_error(seastar::logger&, std::basic_string_view<char, std::char_traits<char> >) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/on_internal_error.cc:80

service::storage_service::handle_state_normal(gms::inet_address, utils::tagged_uuid<gms::permit_id_tag>) at /jenkins/workspace/scylla-master/scylla-ci/scylla/service/storage_service.cc:2513
std::__n4861::coroutine_handle<seastar::internal::coroutine_traits_base<void>::promise_type>::resume() const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/coroutine:242
 (inlined by) seastar::internal::coroutine_traits_base<void>::promise_type::run_and_dispose() at ./build/RelWithDebInfo/seastar/./seastar/include/seastar/core/coroutine.hh:122
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:2606
seastar::reactor::run_some_tasks() at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3078
seastar::reactor::do_run() at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3246
seastar::reactor::run() at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/reactor.cc:3136
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/app-template.cc:276
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at ./build/RelWithDebInfo/seastar/./build/RelWithDebInfo/seastar/./seastar/src/core/app-template.cc:167
scylla_main(int, char**) at /jenkins/workspace/scylla-master/scylla-ci/scylla/main.cc:778
std::function<int (int, char**)>::operator()(int, char**) const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:591
main at /jenkins/workspace/scylla-master/scylla-ci/scylla/main.cc:2391
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x2a087 
/jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/21861/libreloc/libc.so.6 0x2a14a 

```

**Comment by mykaul** [comment]
@gleb-cloudius - can you take a look at the above?

**Comment by gleb-cloudius** [comment]
Now the cause looks different and much harder to reproduce. The crash happens when a node is restarted with new IP, is decommissioned and only then process old gossiper entry with its old IP from other nodes. Looks like the bug was always there, but was hidden by decommission been slow. One way to fix it is to mark the test as `@pytest.mark.no_boot_speedups` but then it hits https://github.com/scylladb/scylladb/issues/22045 (although curiously this is only reported as an error and not test failure). But I think I have a way to ignore this old gossiper entry.

**Comment by bhalevy** [comment]
@scylladb/scylla-maint: why wasn't this issue closed with https://github.com/scylladb/scylladb/pull/21956?

**Comment by gleb-cloudius** [comment]
> [@scylladb/scylla-maint](https://github.com/orgs/scylladb/teams/scylla-maint): why wasn't this issue closed with [#21956](https://github.com/scylladb/scylladb/pull/21956)? [@gleb-cloudius](https://github.com/gleb-cloudius) can the issue be closed now?

It was re-opened because it is till happening, though for another reason. The fix is on the ML "gossiper: ignore gossiper entries with local host id in gossiper mode as well"

**Comment by bhalevy** [comment]
> > [@scylladb/scylla-maint](https://github.com/orgs/scylladb/teams/scylla-maint): why wasn't this issue closed with [#21956](https://github.com/scylladb/scylladb/pull/21956)? [@gleb-cloudius](https://github.com/gleb-cloudius) can the issue be closed now?
> 
> It was re-opened because it is till happening, though for another reason. The fix is on the ML "gossiper: ignore gossiper entries with local host id in gossiper mode as well"

Ah, yes.  I actually approved it there.
I thought it was the one that already fixed the issue.
Now I realize it's a follow up on the original fix.
